src/modules/shared/forms/engine/models/wizard-engine-irv3-schema.model.ts

```diff
@@ -0,0 +1,701 @@
+          if (itemsFromAnswer) {
+            const conditionalAnswerId =
+              subsection?.steps
+                .flatMap(st => st.questions)
+                .find(q => q.id === itemsFromAnswer)
+                ?.items?.find(item => item.conditional)?.conditional?.id ?? '';
+
+            const itemsDependencyAnswer: string[] = this.currentAnswers[itemsFromAnswer]
+              ? (this.currentAnswers[itemsFromAnswer] as arrStringAnswer)
+              : [];
+
+            const updatedItemsList = itemsDependencyAnswer.map(item => ({
+              id: item,
+              label:
+                item === 'OTHER'
+                  ? this.currentAnswers[conditionalAnswerId]
+                  : this.translations.questions.get(itemsFromAnswer)?.items.get(item)?.label
+            }));
+
+            param.items = updatedItemsList;
+
+            if (param.items?.length === 1) {
+              // Hide step, but keep it in 'this.steps' in order to properly add its value to outbound payload
+              param.isHidden = true;
+            }
+          }
+
+          step.parameters.push(param);
+
+          // runtimerules for `addQuestions`
+          const conditionalItem = q.items?.find(i => i.conditional);
+
+          if (q.addQuestion && this.getAnswers()[q.id]) {
+            (this.getAnswers()[q.id] as arrStringAnswer).forEach((answer, i) => {
+              // replace variables on label's placeholders for fields-groups
+              let label = q.addQuestion!.label;
+
+              if (q.dataType === 'fields-group' && q.field) {
+                label = q.addQuestion!.label.replace(
+                  /{{*[^{}]*}*}/,
+                  this.currentAnswers[q.id][i][q.field.id] ?? q.addQuestion!.label
+                );
+              } else if (q.dataType === 'checkbox-array' && q.addQuestion) {
+                const itemAnswer: string =
+                  typeof this.currentAnswers[q.id][i] === 'object'
+                    ? this.currentAnswers[q.id][i][this.getCheckBoxAnswerId(q)]
+                    : this.currentAnswers[q.id][i];
+
+                label = this.translations.questions.get(q.id)?.items.get(itemAnswer)?.label ?? '';
+
+                // search for conditional on question items, then check if answer is from a conditional. If so, get conditional answer
+                if (conditionalItem?.conditional && conditionalItem.id === itemAnswer) {
+                  label = this.currentAnswers[conditionalItem.conditional.id];
+                }
+
+                // replace variable (i.e. '{{item}}') on label's placeholders for checkbox-arrays with addQuestions
+                label = q.addQuestion.label.replace(/{{*[^{}]*}*}/, label);
+              }
+
+              addSteps.push(
+                new FormEngineModelV3({
+                  parameters: [
+                    {
+                      id: `${q.addQuestion!.id}_${i}`,
```

Comment:
It is indeed inside the if that checks for addQuestion, but my guess is TS can't recognize it since at this point we are inside a forEach loop. 
Will wrap the `addSteps.push()` it inside another if